NuRadioReco/modules/channelIceThermalNoiseAdder.py

```diff
@@ -0,0 +1,263 @@
+                            detector.get_antenna_model(station.get_id(), channel.get_id()),
+                            )
+                        antenna_orientation = detector.get_antenna_orientation(station.get_id(), channel.get_id())
+
+                        # add random polarizations and phase to electric field
+                        if self.debug:
+                            polarizations = np.zeros(len(efield_amplitude))
+                        else:
+                            polarizations = np.random.uniform(0, 2 * np.pi, len(efield_amplitude))
+
+                        channel_noise_spec[1][passband_filter] = noise_spectrum[1][passband_filter] * np.cos(polarizations)
+                        channel_noise_spec[2][passband_filter] = noise_spectrum[2][passband_filter] * np.sin(polarizations)
+
+                        # fold electric field with antenna response
+                        antenna_response = self.get_cached_antenna_response(
+                            antenna_pattern, theta, phi, *antenna_orientation)
+
+                        channel_noise_spectrum = (
+                            antenna_response['theta'] * channel_noise_spec[1]
+                            + antenna_response['phi'] * channel_noise_spec[2]
+                        )
+
+                        # add noise spectrum from pixel in the sky to channel spectrum
+                        channel_spectra[channel.get_id()] += channel_noise_spectrum
+
+        # store the updated channel spectra
+        for channel in station.iter_channels():
+            channel.set_frequency_spectrum(channel_spectra[channel.get_id()], "same")
+
+
+if __name__ == "__main__":
+    import datetime
+    import matplotlib.pyplot as plt
+    from NuRadioReco.detector.RNO_G.rnog_detector import Detector
+    from NuRadioReco.framework.event import Event
+    from NuRadioReco.framework.station import Station
+    from NuRadioReco.framework.channel import Channel
+
+    station_id = 23
+    nr_samples = 2048
+    sampling_rate = 3.2 * units.GHz
+    frequencies = np.fft.rfftfreq(nr_samples, d=1./sampling_rate)
+    channel_ids = [0]
+
+    detector = Detector(database_connection='RNOG_public', log_level=logging.NOTSET,
+                        select_stations=station_id)
+    detector_time = datetime.datetime(2023, 8, 1)
+    detector.update(detector_time)
+
+    event = Event(run_number=-1, event_id=-1)
+    station = Station(station_id)
+    station.set_station_time(detector.get_detector_time())
+    for channel_id in channel_ids:
+        channel = Channel(channel_id)
+        channel.set_frequency_spectrum(np.zeros_like(frequencies, dtype=np.complex128), sampling_rate)
+        station.add_channel(channel)
+    event.set_station(station)
+
+    # These files were generated using the code NuRadioMC/examples/simulate_effective_ice_temperature
+    # see the README there for more details
+    eff_temperature_dir = "NuRadioReco/examples/RNOG/thermal_noise/eff_temperatures"
```

Comment:
not sure this will work when running from outside the NuRadioMC directory and/or without the NuRadioMC directory in the PATH... might it be better to replace `NuRadioReco` by the content of the `NuRadioReco.__path__` variable?